### PR TITLE
Add lower bound for translog flush threshold

### PR DESCRIPTION
**Pull request by dnhatn**
If the translog flush threshold is too small (eg. smaller than the
translog header), we may repeatedly flush even there is no uncommitted
operation because the shouldFlush condition can still be true after
flushing. This is currently avoided by adding an extra guard against the
uncommitted operations. However, this extra guard makes the shouldFlush
complicated. This commit replaces that extra guard by a lower bound for
translog flush threshold. We keep the lower bound small for convenience
in testing.

Relates #28350
Relates #23779

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java [hunk 1]
```diff
@@ -1477,14 +1477,7 @@ public boolean shouldPeriodicallyFlush() {
...
         final long uncommittedSizeOfNewCommit = translog.sizeOfGensAboveSeqNoInBytes(localCheckpointTracker.getCheckpoint() + 1);
-        /*
-         * If flushThreshold is too small, we may repeatedly flush even there is no uncommitted operation
-         * as #sizeOfGensAboveSeqNoInByte and #uncommittedSizeInBytes can return different values.
-         * An empty translog file has non-zero `uncommittedSize` (the translog header), and method #sizeOfGensAboveSeqNoInBytes can
-         * return 0 now(no translog gen contains ops above local checkpoint) but method #uncommittedSizeInBytes will return an actual
-         * non-zero value after rolling a new translog generation. This can be avoided by checking the actual uncommitted operations.
-         */
-        return uncommittedSizeOfNewCommit < uncommittedSizeOfCurrentCommit && translog.uncommittedOperations() > 0;
+        return uncommittedSizeOfNewCommit < uncommittedSizeOfCurrentCommit;
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java]: can we add an assertion that the uncommitOps is >0 ?

File: server/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 2]
```diff
@@ -219,9 +226,9 @@
                      * generation threshold. However, small thresholds are useful for testing so we
                      * do not add a large lower bound here.
                      */
-                    new ByteSizeValue(64, ByteSizeUnit.BYTES),
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/index/IndexSettings.java]: I think we need to adapt the comment above?

**Review by bleskes — APPROVED** [review]
LGTM. Left some asks, no need for another review.

**Comment by dnhatn** [comment]
Thanks @bleskes for reviewing.

**Pull request merged**